0 context in this patch to avoid pulling in device-specific strings
--- a/META-INF/com/google/android/updater-script
+++ b/META-INF/com/google/android/updater-script
@@ -8 +8,4 @@
-mount("ext4", "EMMC", "/dev/block/platform/msm_sdcc.1/by-name/system", "/system");
+package_extract_file("dualboot.sh", "/tmp/dualboot.sh");
+set_perm(0, 0, 0777, "/tmp/dualboot.sh");
+ui_print("Installing ROM as secondary");
+run_program("/tmp/dualboot.sh", "mount-system", "ext4", "/dev/block/platform/msm_sdcc.1/by-name/system");
@@ -14 +17 @@
-unmount("/system");
+run_program("/tmp/dualboot.sh", "unmount-system");
@@ -16 +19 @@
-format("ext4", "EMMC", "/dev/block/platform/msm_sdcc.1/by-name/system", "0", "/system");
+ui_print("Formatting /system/dual for dual boot");
@@ -17,0 +21,3 @@
+delete_recursive("/system/dual");
+unmount("/system");
+run_program("/tmp/dualboot.sh", "mount-system", "ext4", "/dev/block/platform/msm_sdcc.1/by-name/system");
@@ -156 +162,2 @@
-unmount("/system");
+run_program("/tmp/dualboot.sh", "set-secondary-kernel");
+run_program("/tmp/dualboot.sh", "unmount-system");
